pallets/creator-staking/src/types/mod.rs

```diff
@@ -0,0 +1,202 @@
+use codec::{Decode, Encode, HasCompact, MaxEncodedLen};
+use frame_support::traits::Currency;
+use scale_info::TypeInfo;
+use sp_arithmetic::traits::AtLeast32BitUnsigned;
+use sp_runtime::{Perbill, traits::Zero, RuntimeDebug};
+use sp_std::{ops::Add, prelude::*};
+
+use super::*;
+
+pub mod impls;
+
+pub(crate) type BalanceOf<T> =
+<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+/// Counter for the number of eras that have passed.
+pub type EraIndex = u32;
+
+// This represents the max assumed vector length that any storage item should have.
+// In particular, this relates to `UnbondingInfo` and `StakerInfo`.
+// In structs which are bound in size, `MaxEncodedLen` can just be derived but that's not the case
+// for standard `vec`. To fix this 100% correctly, we'd need to do one of the following:
+//
+// - Use `BoundedVec` instead of `Vec` and do storage migration
+// - Introduce a new type `S: Get<u32>` into the aforementioned structs and use it to inject max
+//   allowed size, thus allowing us to correctly calculate max encoded len
+//
+// The issue with first approach is that it requires storage migration which we want to avoid
+// unless it's really necessary. The issue with second approach is that it makes code much more
+// difficult to work with since all of it will be ridden with injections of the `S` type.
+const MAX_ASSUMED_VEC_LEN: u32 = 10;
+
+// TODO: there are few more options to chose from:
+//     - UnregistrationSource: This emphasizes where the unregistration is coming from.
+//     - UnregisteringParty: This focuses on who is performing the unregistration.
+//     - UnregistrationActor: This could signify the entity taking action in the unregistration process.
+//     - UnregistrationInitiator: This implies the entity that started or initiated the unregistration.
+/// This enum is used to determine who is calling the `unregister_creator` function.
+pub(super) enum UnregistrationAuthority<AccountId> {
+    Creator(AccountId),
+    Root,
+}
+
+/// Creator State descriptor
+#[derive(Copy, Clone, PartialEq, Eq, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub(super) enum CreatorState {
+    /// Creator is registered and active.
+    Registered,
+    /// Creator has been unregistered and is inactive.
+    /// Claim for past eras and unbonding is still possible but no additional staking can be done.
+    Unregistered(EraIndex),
+}
+
+#[derive(Clone, PartialEq, Eq, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct CreatorInfo<AccountId> {
+    /// Space owner account
+    pub(super) stakeholder: AccountId,
```

Comment:
I think it should be named `beneficiary`.

Reply:
Ask beef

Reply:
beneficiary